### ISSUE TITLE
Optionally save the given Git Access Token when performing a privileged Clone/Pull

### ISSUE THREAD

**Issue by brandonsturgeon** [wisp-gg/frontend#330]
### Describe the feature and why you would like to see it.

When cloning a Private Repository, the User is prompted to enter a Personal Access Token.

However, when you go to pull updates to that Repository, you're again prompted for a Personal Access Token.
This is a super inconvenient and, in my opinion, an unrealistic workflow.

I think that the user should be given the option to Save the given Personal Access Token such that future pulls will succeed without prompting the user (or, if they do prompt the user, give them an easy way to select a previously-used token)

### Describe the solution you'd like.

For starters, whenever the user is prompted for a Personal Access Token, they should also be prompted with a checkbox for "Save this Token for future use?"

The mechanism that actually stores the token could, in my eyes, be done in one of two ways:

### 1. **Include the PAT in the Git URL when cloning**
So this would simply clone the given URL with the token included in the URL: `https://username:token@github.com/org/project`, which would then save the token-ized URL in the `.git/config` file. Future pulls would use that authenticated URL.

#### ✅ **Pros**
 - Easy to implement
 - Easy to use

#### ❌ **Cons**
 - Technically insecure to have the token exposed in plain text. There could be multiple ways to mitigate this, but it's still going to be a problem.
 - Kind of complicated to change/update the token

### 2. **Store the Users's PAT in an encrypted secrets DB**
I imagine there's some place that Wisp stores secure data at the moment. Perhaps a users' PATs could be included in there, and would allow them to easily select already-saved tokens for future use. This could be done manually or automatically.

#### ✅ **Pros**
 - Much more secure
 - Easy to change/edit in the future
 - Equally easy to use

#### ❌ **Cons**
 - Harder, more time consuming to implement
 - Some software and frontend design decisions to consider

**Comment by TrixterTheTux** [comment]
Option 1 is already possible as you can include authentication credentials in the clone URL.

**Comment by brandonsturgeon** [comment]
Right, and that's what I currently do, but it's a bit of an arcane process for a user who just wants to clone a private repo.

Wisp can already ask for the token, I'm just requesting/suggesting that Wisp assemble the final URL for them.
